x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLClientAuthTests.java

```diff
@@ -54,13 +58,41 @@ protected boolean addMockHttpTransport() {
 
     @Override
     protected Settings nodeSettings(int nodeOrdinal) {
-        return Settings.builder()
-                .put(super.nodeSettings(nodeOrdinal))
+        Settings baseSettings = super.nodeSettings(nodeOrdinal);
+
+        Settings.Builder builder = Settings.builder().put(baseSettings);
+        baseSettings.getByPrefix("xpack.security.transport.ssl.")
+                .keySet()
+                .forEach(k -> {
+                    String httpKey = "xpack.security.http.ssl." + k;
+                    String value = baseSettings.get("xpack.security.transport.ssl." + k);
+                    if (value != null) {
+                        builder.put(httpKey, baseSettings.get("xpack.security.transport.ssl." + k));
+                    }
+                });
+
+        MockSecureSettings secureSettings = (MockSecureSettings) builder.getSecureSettings();
+        for (String key : new HashSet<>(secureSettings.getSettingNames())) {
+            SecureString value = secureSettings.getString(key);
+            if (value == null) {
+                try {
+                    if (key.startsWith("xpack.security.transport.ssl.")) {
+                        byte[] file = Streams.readAll(secureSettings.getFile(key));
```

Comment:
Maybe move [`toBytesArray`](https://github.com/elastic/elasticsearch/blob/9e529d9d584ae6f6978ac09056b73d9ccdde4a3c/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/esnative/tool/CommandLineHttpClientTests.java#L102) to ESTestCase and use this instead so that we don't introduce BC dependencies again?